### PR TITLE
Validate colors restored from settings; correct and replace as required.

### LINKED ISSUE #511: Adding an extra color to the theme field in dconf breaks the terminal
If you accidentally add an extra color to the theme via dconf-editor, it crashes the terminal and stops it from opening again. It should either truncate the parameters, not let you save, or ignore the extra values.

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #511

### CODE REVIEW

File: src/Widgets/TerminalWidget.vala [hunk 1]
```diff
@@ -247,32 +247,54 @@ namespace Terminal {
...
-                    current_string += Application.settings.get_string ("palette")[i].to_string ();
-                }
+            var hex_palette = new string[PALETTE_SIZE];
+            var palette_setting_string = Application.settings.get_string ("palette");
+            var setting_palette = palette_setting_string.split (":", PALETTE_SIZE + 1);
+
+            bool settings_valid = setting_palette.length <= PALETTE_SIZE;
+
+            int i = 0;
+            foreach (string hex in setting_palette) {
```
Comment (tintou) [inline src/Widgets/TerminalWidget.vala:265]:
```suggestion
            foreach (unowned string hex in setting_palette) {
```

File: src/Widgets/TerminalWidget.vala [hunk 2]
```diff
@@ -247,32 +247,54 @@ namespace Terminal {
...
+             /* Remove invalid colors from setting */
+                var sb = new StringBuilder ("");
+                foreach (string hex in hex_palette) {
+                    sb.append (hex);
+                    sb.append (":");
+                }
 
-                palette[i] = new_color;
+                sb.truncate (sb.len - 1);
+                Application.settings.set_string ("palette", sb.str);
```
Comment (tintou) [inline src/Widgets/TerminalWidget.vala:297]: What you want is this :slightly_smiling_face: 
```suggestion
                Application.settings.set_string ("palette", string.joinv (":", hex_palette));
```

File: src/Widgets/TerminalWidget.vala (hunk 1 above)
Comment (jeremypw) [reply src/Widgets/TerminalWidget.vala]: Done

File: src/Widgets/TerminalWidget.vala (hunk 2 above)
Comment (jeremypw) [reply src/Widgets/TerminalWidget.vala]: Much better - thanks.

**Pull request merged**